quiz/src/test/java/rx/quiz/observables/IntermediateCreateIntervalKoanTest.java

```diff
@@ -0,0 +1,26 @@
+package rx.quiz.observables;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.Scheduler;
+import io.reactivex.annotations.NonNull;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.TimeUnit;
+
+public class IntermediateCreateIntervalKoanTest extends IntermediateCreateIntervalKoan {
+
+    @NotNull
+    @Override
+    public Observable<Integer> interval(@NotNull Scheduler scheduler, long period, @NotNull TimeUnit unit) {
+        return Observable.create(emitter -> emitValues(emitter, 0, scheduler, period, unit));
+    }
+
+    private void emitValues(@NonNull ObservableEmitter<Integer> emitter, int value, @NotNull Scheduler scheduler,
+                            long period, @NotNull TimeUnit unit) {
+        scheduler.scheduleDirect(() -> {
+            emitter.onNext(value);
+            emitValues(emitter, value + 1, scheduler, period, unit);
```

Comment:
Alternative solution to this koan is to schedule a Runnable (non-lambda):

scheduler.scheduleDirect(new Runnanle(){
    run(){
        emitter.onNext(value.incrementAndGet());
        scheduler.scheduleDirect(this);
    }
}, delay, unit);

Reply:
Yes, it's much shorter and no need of recursion.